### DISCUSSION TITLE
janusgraph health checks

### DISCUSSION THREAD

**Discussion by krittamk** [JanusGraph/janusgraph#2621] [Q&A]
I have a janusgraph server running as a dedicated gremlin server.
To connect to it i'm using another service as a client which uses the java connector.

     traversalSource = traversal().withRemote(propertiesFile);

Occasionally i get exceptions like All hosts are considered unavailable due to previous exceptions. Check the error log to find the actual reason.
I need to setup a health check in this client service such that whenever no hosts are available, the health check automatically fails.
What can be used as a safe health check logic ?

**Comment by vtslab** [comment]
OK that you reach out to be sure to have the idiomatic solution. I am not aware of any API you could use instead of trying to connect. So, you can just wrap this line in a try/catch block and you will be fine.
Also, the exception suggests to check the error log. It should be possible to automate these checks with tools such as Prometheus, so that error situations can be solved before clients are affected.

**Comment by farodin91** [comment]
I think it should be possible to add a good healthcheck as part of the grpc. I don't expect this near future.
